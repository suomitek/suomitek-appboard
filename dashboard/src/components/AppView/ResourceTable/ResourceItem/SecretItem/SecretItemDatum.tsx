import * as React from "react";
import { Eye, EyeOff } from "react-feather";

interface ISecretItemDatumProps {
  name: string;
  value: string;
}

interface ISecretItemDatumState {
  hidden: boolean;
}

class SecretItemDatum extends React.PureComponent<ISecretItemDatumProps, ISecretItemDatumState> {
  // Secret datum is hidden by default
  public state: ISecretItemDatumState = {
    hidden: true,
  };

  public render() {
    const { name, value } = this.props;
    const { hidden } = this.state;
    const decodedValue = atob(value);
    return (
      <span className="flex">
        <button className="secretDatum" onClick={this.toggleDisplay}>
          {hidden ? <Eye /> : <EyeOff />}
        </button>
        <span className="flex margin-l-normal">
          <span>{name}:</span>
          {hidden ? (
            <span className="margin-l-small">{decodedValue.length} bytes</span>
          ) : (
            <pre className="SecretContainer">
              <code className="SecretContent">{decodedValue}</code>
            </pre>
          )}
        </span>
      </span>
    );
  }

  private toggleDisplay = () => {
    this.setState({
      hidden: !this.state.hidden,
    });
  };
}

export default SecretItemDatum;
