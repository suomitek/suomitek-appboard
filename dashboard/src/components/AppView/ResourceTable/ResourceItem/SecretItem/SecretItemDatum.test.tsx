import { shallow } from "enzyme";
import * as React from "react";

import SecretItemDatum from "./SecretItemDatum";

const testProps = {
  name: "foo",
  value: "YmFy", // foo
};

it("renders the secret datum (hidden by default)", () => {
  const wrapper = shallow(<SecretItemDatum {...testProps} />);
  expect(wrapper.state()).toMatchObject({ hidden: true });
  expect(wrapper).toMatchSnapshot();
});

it("displays the secret datum value when clicking on the icon", () => {
  const wrapper = shallow(<SecretItemDatum {...testProps} />);
  expect(wrapper.text()).toContain("foo:3 bytes");
  const icon = wrapper.find("button");
  expect(icon).toExist();
  icon.simulate("click");
  expect(wrapper.state()).toMatchObject({ hidden: false });
  expect(wrapper.text()).toContain("foo:bar");
});
