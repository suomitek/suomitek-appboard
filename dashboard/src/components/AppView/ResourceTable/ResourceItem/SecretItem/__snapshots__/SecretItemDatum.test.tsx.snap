// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the secret datum (hidden by default) 1`] = `
<span
  className="flex"
>
  <button
    className="secretDatum"
    onClick={[Function]}
  >
    <Eye
      color="currentColor"
      size="24"
    />
  </button>
  <span
    className="flex margin-l-normal"
  >
    <span>
      foo
      :
    </span>
    <span
      className="margin-l-small"
    >
      3
       bytes
    </span>
  </span>
</span>
`;
